homeassistant/components/template/light.py

```diff
@@ -335,3 +390,90 @@ def template_light_startup(event):
                         self._name,
                         ex,
                     )
+
+    @callback
+    def update_temperature(self):
+        """Update the temperature from the template."""
+        if self._temperature_template is not None:
+            try:
+                temperature = int(self._temperature_template.async_render())
+            except TemplateError:
+                _LOGGER.error("Cannot evaluate temperature template", exc_info=True)
+                self._temperature = None
+
+            if self.min_mireds <= temperature <= self.max_mireds:
+                self._temperature = temperature
+            else:
+                _LOGGER.error(
+                    "Received invalid color temperature : %s. Expected: 0-%s",
+                    temperature,
+                    self.max_mireds,
+                )
+                self._temperature = None
+
+    @callback
+    def update_brightness(self):
+        """Update the brightness from the template."""
+        if self._level_template is not None:
+            try:
+                brightness = self._level_template.async_render()
+            except TemplateError as ex:
+                _LOGGER.error(ex)
+                self._state = None
+
+            if 0 <= int(brightness) <= 255:
+                self._brightness = int(brightness)
+            else:
+                _LOGGER.error(
+                    "Received invalid brightness : %s. Expected: 0-255", brightness
+                )
+                self._brightness = None
+
+    @callback
+    def update_color(self):
+        """Update the hs_color from the template."""
+        if self._color_template is not None:
+            h_str = None
+            s_str = None
+            self._color = None
+
+            try:
+                render = self._color_template.async_render()
+                h_str, s_str = map(
+                    int, render.replace("(", "").replace(")", "").split(",", 1)
```

Comment:
Should be float

Reply:
From a hue lamp 
![FireShot Capture 032 - Home Assistant - hassio lan](https://user-images.githubusercontent.com/45088785/72120717-2d19e880-3359-11ea-87f9-3dfa514c8fdb.png)

Reply:
Oh I didn't know. I will change this.